torch/csrc/jit/register_string_ops.cpp

```diff
@@ -547,48 +547,94 @@ auto reg_str_ops_2 = torch::RegisterOperators()
            post_partition);
         })
 
-    .op("aten::split(str self, str separator=' ', int max=-1) -> str[]",
+    // Due to type constraints in TorchScript, we use a sentinel string
+    // "__DO_NOT_USE" as the default separator for split functions.
+    // This denotes splitting by any whitespace, or default behavior when
+    // the separator is not specified
+    .op("aten::split(str self, str separator='__DO_NOT_USE', int max=-1) -> str[]",
```

Comment:
Can `separator` be type `str?` instead?

Reply:
ooh, thanks! didn't realize this existed -- cleans up a lot